plugins/modules/cloud/misc/terraform.py

```diff
@@ -443,7 +443,7 @@ def main():
     for k, v in variables.items():
         variables_args.extend([
             '-var',
-            '{0}={1}'.format(k, v)
+            '{0}={1}'.format(k, json.dumps(v))
```

Comment:
How does terraform interpret arguments of `-var`? I'm wondering whether this is a breaking change or not.

Reply:
Could you please comment on this one?

Reply:
I don't believe so, as I haven't hit any issues with this patch in my use cases. That said, I just reread the TF docs on the topic (see [Input Variables on the Command Line](https://www.terraform.io/cli/commands/plan#input-variables-on-the-command-line) and [Assigning Values to Root Module Variables](https://www.terraform.io/language/values/variables#assigning-values-to-root-module-variables)), and the references all seem to point to TF "native" expressions. However, [JSON Configuration Syntax](https://www.terraform.io/language/syntax/json) states that _"Everything that can be expressed in native syntax can also be expressed in JSON syntax, but some constructs are more complex to represent in JSON due to limitations of the JSON grammar."_ 

I'll run some more tests to see if I can break this, but I would think that if you have _really_ complex variables, you are best served by creating your own variables file and passing that as a reference. The intention of this patch is to allow "simple" complex variables, like `list(string)`, to be declared in the `variables` parameter.

Reply:
Thanks for your explanations and links! Sounds convincing to me. (I'm not using terraform so I can't test myself.)

Reply:
This has turned out to be a breaking change. See #4367.